### ISSUE TITLE
Better to return the raw error message returned from Softlayer API

### ISSUE THREAD

**Issue by mattcui** [maximilien/softlayer-go#101]
For example, If I tried to create a public image based on an non-existing image with CreatePublicArchiveTransaction function:

```
POST /rest/v3/SoftLayer_Virtual_Guest_Block_Device_Template_Group/111111/createPublicArchiveTransaction.json HTTP/1.1
Host: api.softlayer.com

{"parameters":["light-stemcell-test-1","Public+light+stemcell","Public+light+stemcell",[265592,358694,449494]]}
[softlayer-go] Response:
HTTP/1.1 404 Not Found
Connection: close
Content-Length: 98
Content-Type: application/json
Date: Sun, 08 May 2016 12:42:54 GMT
Server: Apache
Vary: Accept-Encoding
X-Frame-Options: SAMEORIGIN

{"error":"Unable to find object with id of '111111'.","code":"SoftLayer_Exception_ObjectNotFound"}
```
But err.Error() returned the follow message:
```
softlayer-go: could not SoftLayer_Virtual_Guest_Block_Device_Template_Group#createPublicArchiveTransaction, HTTP error code: '404'
```
I would like to see err.Error() could return "Unable to find object with id of '111111'." in this case, I need to parse the error message to take different subsequent action. Thanks.

**Comment by maximilien** [comment]
Would keeping track of previous error, like wrapping errors as they do in BOSH work?

**Comment by mattcui** [comment]
Just need to return what error info returned by Softlayer API, not much helpful with the info returned from Softlayer-go api currently, like that "could not SoftLayer_Virtual_Guest_Block_Device_Template_Group#createPublicArchiveTransaction"

**Comment by maximilien** [comment]
Well that's also useful since that tells me where the error occurred. I think wrapping like they do in BOSH is the right approach

**Comment by TheKevJames** [comment]
This should be fixed by #136 , I believe.

As a side note; I just got bit by this problem since terraform [still has v0.6 bundled](https://github.com/hashicorp/terraform/blob/master/vendor/vendor.json#L1294). Any chance we can get a new release for bundling into terraform?

**Comment by maximilien** [comment]
@TheKevJames, correct. Closing and also doing a new release now. We've tested this and a few more fixes and seems good. Thanks for contribution :cheers:
